docs/credhub-security-faq.md

```diff
@@ -52,12 +52,12 @@ TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384
 
 If CredHub's third-party integration with a Luna SafeNet HSM is
 enabled, then CredHub has no knowledge of key values. If CredHub is
-started in internal encryption mode, CredHub deterministically
-generates a site-specific AES256 key on startup. CredHub concatenates
-a user-defined password from its configuration file with a
+started in internal encryption mode, CredHub recalculates
+the AES256 key on startup. To do this, CredHub concatenates
+the user-provided encryption password from its configuration file with a
 randomly-generated salt stored in its database and hashes the
```

Comment:
It's not clear from this when CredHub is randomly generating the salt. Could be at every startup from what this says.

Reply:
OK. To me, "stored in its database" was enough to explain that it is only generated the first time.

I'm going to rewrite the whole thing, and split it in 2 sections:
1. what happens the first time
2. what happens on the next restarts of the server

that should make everything clearer.

Reply:
For what it's worth, the other edits look fine.